users/models.py

```diff
@@ -91,8 +92,20 @@ class Profile(models.Model):
     post_karma = models.IntegerField(default=0)
     gold_awards = models.IntegerField(default=0)
     gender = models.CharField(choices=GENDER_CHOICES, max_length=1)
-    avatar = models.ImageField(upload_to=user_avatar_path, null=True, blank=True, default=None)
-    banner = models.ImageField(upload_to=user_banner_path, null=True, blank=True, default=None)
+    avatar = models.ImageField(
+        upload_to=user_avatar_path,
+        null=True,
+        blank=True,
+        default=None,
+        validators=[FileExtensionValidator(allowed_extensions=["jpg", "jpeg", "png"])],
+    )
+    banner = models.ImageField(
+        upload_to=user_banner_path,
+        null=True,
+        blank=True,
+        default=None,
+        validators=[FileExtensionValidator(allowed_extensions=["jpg", "jpeg", "png"])],
```

Comment:
1. Please set those extensions in settings.py
2. Also you should update template in all places: https://github.com/pymasterspl/reddit/blob/95634a4ef648a6650404b15a38e28f489ea76263/users/templates/users/profile_settings.html#L36. Right now there is discrepancy in extensions. 
3. How can you verify the file is really an image, not some kind of zipped executable?
![image](https://github.com/user-attachments/assets/511119b1-eec6-4af4-8ab1-c875346bd060)
https://docs.djangoproject.com/en/5.1/ref/validators/#fileextensionvalidator 
So sadly, FileExtensionValidator is not enough. 

https://pythoncircle.com/post/773/how-to-validate-an-uploaded-image-in-python-django/ - I don't know how old is this, but shows couple ideas.